torch/_dynamo/variables/builder.py

```diff
@@ -689,12 +689,12 @@ def index_source(key):
                 source=self.source,
             )
         elif trace_rules.lookup(value) is not None:
+            if is_user_defined_allowed(value):
+                self.tx.output.has_user_defined_allowed_in_graph = True
             return trace_rules.lookup(value).create_with_source(
                 value, source=self.source
             )
         elif is_allowed(value):
-            if is_user_defined_allowed(value):
-                self.tx.output.has_user_defined_allowed_in_graph = True
```

Comment:
What's going on here?

Reply:
I moved it 6 lines up to the ```elif trace_rules.lookup(value) is not None:``` branch, since this is to handle user defined allow in graph function, which would covered by ```lookup``` branch.
Actually we supposed to support ```allow_in_graph``` to decorate ```nn.Module``` and ```autograd.Function```, but these two cases are not working well now, #113510 is going to fix the ```autograd.Function``` thing, I'll have another PR to fix the ```nn.Module``` thing.